misc/docker-ci/Dockerfile.ubuntu2004

```diff
@@ -41,10 +39,60 @@ RUN apt-get install --yes \
+		&& ln -vnf $(find target/release -name libcrypto.a -o -name libssl.a) quiche/deps/boringssl/src/lib/ \
+		&& find . -name '*.so' -delete) \
+	&& (wget --https-only --no-verbose -O - "https://curl.se/download/curl-${CURL_VERSION}.tar.bz2" | tar xjf - \
+		&& cd "curl-${CURL_VERSION}" \
+		&& USE_CURL_SSLKEYLOGFILE=true ./configure --prefix="${CURL_INSTALL_DIR}" --with-nghttp2 --disable-shared  CC=clang CXX=clang++ LDFLAGS="-Wl,-rpath,$PWD/../quiche/target/release" --with-openssl="$PWD/../quiche/quiche/deps/boringssl/src" --with-quiche="$PWD/../quiche/target/release" --program-suffix="-quiche" \
+		&& make -j"$(nproc)" V=1 && make -j"$(nproc)" V=1 install) \
+	&& (rm -rf quiche "curl-${CURL_VERSION}") \
+	&& curl-quiche --version \
+	&& curl-quiche -sSfv --http3 https://http3.is > /dev/null
+
+# curl-ngtcp2 (HTTP/3 using ngtcp2+nghttp3 and a patched OpenSSL)
+RUN (git clone --depth 1 --branch openssl-3.0.3+quic https://github.com/quictls/openssl \
+		&& cd openssl \
+		&& CC=clang CXX=clang++ ./config enable-tls1_3 --prefix="${HOME}/curl" \
+		&& make -j"$(nproc)" V=1 && make -j"$(nproc)" V=1 install) \
+	&& (git clone --depth 1 --branch v0.4.1 https://github.com/ngtcp2/nghttp3 \
+		&& cd nghttp3 \
+		&& autoreconf -fi \
+		&& ./configure CC=clang CXX=clang++ PKG_CONFIG_PATH="${HOME}/curl/lib/pkgconfig:${HOME}/curl/lib64/pkgconfig" LDFLAGS="-Wl,-rpath,${HOME}/curl/lib:${HOME}/curl/lib64" --prefix="${HOME}/curl" --enable-lib-only --disable-shared \
+		&& make -j"$(nproc)" V=1 && make -j"$(nproc)" V=1 install) \
+	&& (git clone --depth 1 --branch v0.5.0 https://github.com/ngtcp2/ngtcp2 \
+		&& cd ngtcp2 \
+		&& autoreconf -fi \
+		&& ./configure CC=clang CXX=clang++ PKG_CONFIG_PATH="${HOME}/curl/lib/pkgconfig:${HOME}/curl/lib64/pkgconfig" LDFLAGS="-Wl,-rpath,${HOME}/curl/lib:${HOME}/curl/lib64" "--prefix=${HOME}/curl" --enable-lib-only --disable-shared \
+		&& make -j"$(nproc)" V=1 && make -j"$(nproc)" V=1 install) \
+	&& (find "${HOME}/curl" -name '*.so*' -delete) \
+	&& (wget --https-only --quiet -O - "https://curl.se/download/curl-${CURL_VERSION}.tar.bz2" | tar xjf - \
+		&& cd "curl-${CURL_VERSION}" \
+		&& USE_CURL_SSLKEYLOGFILE=true ./configure CC=clang CXX=clang++ LDFLAGS="-Wl,-rpath,${HOME}/curl/lib:${HOME}/curl/lib64" --prefix="${CURL_INSTALL_DIR}" --with-nghttp2 --disable-shared --with-openssl="${HOME}/curl" --with-ngtcp2="${HOME}/curl" --with-nghttp3="${HOME}/curl" --program-suffix="-ngtcp2" \
+		&& make -j"$(nproc)" V=1 && make -j"$(nproc)" V=1 install) \
+	&& (rm -rf openssl nghttp3 ngtcp2 "curl-${CURL_VERSION}" curl) \
+	&& curl-ngtcp2 --version \
+	&& curl-ngtcp2 -sSfv --http3 https://http3.is > /dev/null \
+	&& ln -s "${CURL_INSTALL_DIR}/bin/curl-ngtcp2" "${CURL_INSTALL_DIR}/bin/curl"
+
+# libssl-dev causes curl to fail to build with http3 (quiche).
+# Once curl is built with BoringSSL's static libraries, libssl-dev can be installed without affecting curl.
+RUN apt-get install --yes libssl-dev
```

Comment:
Changes made here looks good.

Yes, I think it makes sense to have these on Ubuntu 20.04 rather than 22.04, because 20.04 is the one that we run multiple flavors of our CI (OpenSSL 1.1.0, 1.1.1, asan), and we want run http3 tests on as much flavors as possible.